providers/openebs/installers/operator/master/ansible/openebs_setup.yaml

```diff
@@ -60,79 +60,74 @@
+          shell: "{{ kubeapply }} set env deployment maya-apiserver \"OPENEBS_IO_JIVA_REPLICA_IMAGE={{ lookup('env','OPENEBS_IO_JIVA_REPLICA_IMAGE') }}\" -n {{ namespace }}"
+          args:
+            executable: /bin/bash
           when: lookup('env','OPENEBS_IO_JIVA_REPLICA_IMAGE') | length > 0
 
-        - name: Applying openebs operator
-          shell: "{{ kubeapply }} apply -f {{ openebs_operator }}"
+        - name: Updating openebs jiva replica count
+          shell: "{{ kubeapply }} set env deployment maya-apiserver \"OPENEBS_IO_JIVA_REPLICA_COUNT={{ lookup('env','OPENEBS_IO_JIVA_REPLICA_COUNT') }}\" -n {{ namespace }}"
           args:
             executable: /bin/bash
+          when: lookup('env','OPENEBS_IO_JIVA_REPLICA_COUNT') | length > 0
 
-        - name: Applying storageclasses
-          shell: "{{ kubeapply }} apply -f {{ storageclass }}"
+        - name: Updating openebs cstor target image
+          shell: "{{ kubeapply }} set env deployment maya-apiserver \"OPENEBS_IO_CSTOR_TARGET_IMAGE={{ lookup('env','OPENEBS_IO_CSTOR_TARGET_IMAGE') }}\" -n {{ namespace }}"
+          args:
+            executable: /bin/bash
+          when: lookup('env','OPENEBS_IO_CSTOR_TARGET_IMAGE') | length > 0
+
+        - name: Updating openebs cstor pool image
+          shell: "{{ kubeapply }} set env deployment maya-apiserver \"OPENEBS_IO_CSTOR_POOL_IMAGE={{ lookup('env','OPENEBS_IO_CSTOR_POOL_IMAGE') }}\" -n {{ namespace }}"
+          args:
+            executable: /bin/bash
+          when: lookup('env','OPENEBS_IO_CSTOR_POOL_IMAGE') | length > 0
+
+        - name: Updating openebs cstor pool management image
+          shell: "{{ kubeapply }} set env deployment maya-apiserver \"OPENEBS_IO_CSTOR_POOL_MGMT_IMAGE={{ lookup('env','OPENEBS_IO_CSTOR_POOL_MGMT_IMAGE') }}\" -n {{ namespace }}"
           args:
             executable: /bin/bash
+          when: lookup('env','OPENEBS_IO_CSTOR_POOL_MGMT_IMAGE') | length > 0
 
+        - name: Updating openebs cstor volume management image
+          shell: "{{ kubeapply }} set env deployment maya-apiserver \"OPENEBS_IO_CSTOR_VOLUME_MGMT_IMAGE={{ lookup('env','OPENEBS_IO_CSTOR_VOLUME_MGMT_IMAGE') }}\" -n {{ namespace }}"
+          args:
+            executable: /bin/bash
+          when: lookup('env','OPENEBS_IO_CSTOR_VOLUME_MGMT_IMAGE') | length > 0
+
+        - name: Updating openebs cstor volume monitor image
+          shell: "{{ kubeapply }} set env deployment maya-apiserver \"OPENEBS_IO_VOLUME_MONITOR_IMAGE={{ lookup('env','OPENEBS_IO_VOLUME_MONITOR_IMAGE') }}\" -n {{ namespace }}"
+          args:
+            executable: /bin/bash
+          when: lookup('env','OPENEBS_IO_VOLUME_MONITOR_IMAGE') | length > 0
+
+        - name: Restarting all pods in openebs namespace
+          shell: "kubectl delete po --all -n {{ namespace }}"
+          args:
+            executable: /bin/bash
 
         - name: Checking Maya-API-Server is running

```

Comment:
https://github.com/openebs/litmus/blob/3e02676b88d68851426c812bc4d76437ed2514a7/providers/openebs/installers/operator/master/ansible/openebs_setup.yaml#L173

@ashishranjan738  Can we also update the logic to get the number of ndm pods running? Currently we have a logic to validate ndm pods against the number of nodes available. Can we change this and validate with the number of untainted nodes?

Reply:
done